src/store/modules/util/actions.ts

```diff
@@ -1,33 +1,128 @@
+import { UtilService } from '@/services/UtilService'
 import { ActionTree } from 'vuex'
 import RootState from '@/store/RootState'
-import * as types from './mutation-types'
-import { hasError } from '@/utils'
 import UtilState from './UtilState'
-import { UtilService } from '@/services/UtilService'
+import * as types from './mutation-types'
+import { showToast } from '@/utils'
+import { hasError } from '@/adapter'
+import { translate } from '@hotwax/dxp-components'
 
 const actions: ActionTree<UtilState, RootState> = {
-  async getSecurityGroups({ commit, state }) {
-    // don't fetch security group information if we already have security groups available in state
-    // Added condition as security group information will not be changed frequently
-    if(state.securityGroups.length) {
-      return;
+  async getRoles({ commit }) {
+    let roles = []
+    const params = {
+      inputFields: {
+        parentTypeId_fld0_value: 'APPLICATION_USER',
+        parentTypeId_fld0_op: 'equals',
+        parentTypeId_fld0_grp: '1',
+        roleTypeId_fld0_value: 'APPLICATION_USER',
+        roleTypeId_fld0_op: 'equals',
+        roleTypeId_fld0_grp: '2',
+      },
+      viewSize: 100,
+      entityName: 'RoleType',
+      fieldList: ['roleTypeId', 'parentTypeId', 'description']
     }
 
-    const payload = {
-      "entityName": "SecurityGroup",
-      "viewSize": 200,
-      "distinct": "Y",
-      "noConditionFind": "Y",
-      "fieldList": ["groupId", "groupName"]
+    try {
+      const resp = await UtilService.getRoles(params)
+      if (!hasError(resp)) {
+        roles = resp.data.docs
+        // pushing none explicitly to show on UI
+        roles.push({
+          roleTypeId: 'none',
+          parentTypeId: 'none',
+          description: 'None',
+        })
+      } else {
+        throw resp.data
+      }
+    } catch (error) {
+      showToast(translate('Something went wrong.'));
+      console.error(error)
+    }
+    commit(types.UTIL_ROLES_UPDATED, roles)
+  },
+
+  async getProductStores({ commit }) {
+    let productStores = []
+    // TODO verify filtering condition for store type
+    const params = {
+      viewSize: 100,
+      noConditionFind: 'Y',
+      entityName: 'ProductStore',
+      fieldList: ['productStoreId', 'storeName']
+    }
+
+    try {
+      const resp = await UtilService.fetchProductStores(params)
+      if (!hasError(resp)) {
+        productStores = resp.data.docs
+      } else {
+        throw resp.data
+      }
+    } catch (error) {
+      showToast(translate('Something went wrong.'));
+      console.error(error)
+    }
+    commit(types.UTIL_PRODUCT_STORES_UPDATED, productStores)
+  },
+
+  async getUserProductStores({ commit, dispatch }, partyId) {
+    let userProductStores = []
+    const params = {
+      inputFields: {
+        partyId,
+      },
+      viewSize: 100,
+      entityName: 'ProductStoreRole',
```

Comment:
Use ProductStoreAndRole entity instead of ProductStoreRole. So at least you need not fetch product stores here.